bin/autodetect/Main.cc

```diff
@@ -212,12 +207,19 @@ int main(int argc, char** argv) {
         mutableFields.push_back(ml::api::CFieldDataCategorizer::MLCATEGORY_NAME);
     }
 
+    const ml::api::CAnomalyJobConfig::CAnalysisConfig& analysisConfig =
+        jobConfig.analysisConfig();
+    const std::string& summaryCountFieldName = analysisConfig.summaryCountFieldName();
+    ml::core_t::TTime bucketSpan = analysisConfig.bucketSpan();
+    ml::core_t::TTime latency = analysisConfig.latency();
+    bool multivariateByFields = analysisConfig.multivariateByFields();
+
     ml::model_t::ESummaryMode summaryMode{
         summaryCountFieldName.empty() ? ml::model_t::E_None : ml::model_t::E_Manual};
     ml::model::CAnomalyDetectorModelConfig modelConfig{ml::model::CAnomalyDetectorModelConfig::defaultConfig(
         bucketSpan, summaryMode, summaryCountFieldName, latency, multivariateByFields)};
     modelConfig.detectionRules(ml::model::CAnomalyDetectorModelConfig::TIntDetectionRuleVecUMapCRef(
-        fieldConfig.detectionRules()));
+        analysisConfig.detectionRules()));
```

Comment:
In the long term this could be encapsulated in a method of `CAnomalyJobConfig`, say `CAnomalyJobConfig::makeModelConfig()`.  But since you're doing a series of PRs you can save this for the next one.